src/parser/declaration.rs

```diff
@@ -0,0 +1,273 @@
+                    }
+                    "border-radius" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::BorderRadius(length))
+                    }
+                    "text-align" => {
+                        parser.consume_colon_separator()?;
+                        let text_al = parser.parse()?;
+                        Ok(Declaration::TextAlign(text_al))
+                    }
+
+                    _ => Err(ParsingError::wrong_token(token_at, "a declaration")),
+                },
+                _ => Err(ParsingError::wrong_token(token_at, "a declaration")),
+            },
+            _ => Err(ParsingError::end_of_file("a declaration")),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn background_color() {
+        let mut parser = Parser::new("background-color: red".chars());
+        assert_eq!(Ok(Declaration::BackgroundColor(Color::Red)), parser.parse());
+        assert_eq!(None, parser.tokens.next());
+    }
+
+    #[test]
+    fn background_color_garbage() {
+        let mut parser = Parser::new("background-color: aifdsn".chars());
+        assert!(parser.parse::<Declaration>().is_err());
+    }
+
+    #[test]
+    fn border_color() {
+        let mut parser = Parser::new("border-color: red".chars());
+        assert_eq!(Ok(Declaration::BorderColor(Color::Red)), parser.parse());
+        assert_eq!(None, parser.tokens.next());
+    }
+
+    #[test]
+    fn border_color_garbage() {
+        let mut parser = Parser::new("border-color: aifdsn".chars());
+        assert!(parser.parse::<Declaration>().is_err());
+    }
+
+    #[test]
+    fn opacity() {
+        let mut parser = Parser::new("opacity: 0.3".chars());
+        assert_eq!(Ok(Declaration::Opacity(0.3)), parser.parse());
+        assert_eq!(None, parser.tokens.next());
+    }
+
+    #[test]
+    fn opacity_negative() {
+        let mut parser = Parser::new("opacity: -4.3".chars());
+        assert!(parser.parse::<Declaration>().is_err());
+    }
+
+    #[test]
+    fn font_family() {
+        let mut parser = Parser::new("font-family: Arial".chars());
+        assert_eq!(
+            Ok(Declaration::FontFamily(FontFamily(
+                vec!["Arial".to_owned()]
+            ))),
+            parser.parse()
+        );
+        assert_eq!(None, parser.tokens.next());
+    }
+
+    #[test]
+    fn text_align() {
+        let mut parser = Parser::new("text-align: center".chars());
+        assert_eq!(
+            Ok(Declaration::TextAlign(TextAlign::Center)),
+            parser.parse()
+        );
+        assert_eq!(None, parser.tokens.next());
+    }
+
+    #[test]
+    fn font_size() {
+        let mut parser = Parser::new("font-size: 3px".chars());
+        assert_eq!(
+            Ok(Declaration::FontSize(Length::Length(
+                3.0,
+                length::LengthUnit::Pixels
```

Comment:
if you want you can do `use length::LengthUnit;` at the beggining of `mod tests {` and then you can just do `LengthUnit::Pixels` instead of `length::LengthUnit::Pixels`